tools/generate.sh

```diff
@@ -1,9 +1,16 @@
 DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 SRC_DIR=$DIR/../src
-ls -1 $DIR/include/ | xargs -n 1 -P $(nproc) -I{} -t bash -c "python3.6 $DIR/te.py $DIR/include/{} | clang-format-5.0 -style=file > $SRC_DIR/include/migraphx/{}"
+PYTHON=python3
+if type -p python3.6 > /dev/null ; then
```

Comment:
Can't we just do something like $PYTHON --version, then parse to make sure it is at least 3.6?

Reply:
Python 3 might not be mapped to 3.6 or 3.8. Also, version comparisons are not simple to do in bash.

Reply:
But currently if python3 is not mapped to 3.6 or 3.8, then we proceed with executing anyways. Are we wanting to version check?

Reply:
This is just to make it work on ubuntu 20.04 which doesn't have python3.6.

Reply:
Yes, previously this script was pinned to python3.6 and ubuntu 20.04 comes with 3.8 by default, so it would fail. But it seems that we want 3.6 minimum and anything greater than that should work? Otherwise, if a user has a version < 3.6, then currently the bash script would still try to execute the python commands.

There is a way to use `sort` to do this that might be better:
```
[ "$( (echo "${MIN_VERSION}" ; echo "${CURR_VERSION}") | sort -V | head -1)" == "${MIN_VERSION}" ]
```

Reply:
Well that doesn't really work, it always outputs the `${MIN_VERSION}`. I think we can leave it for now, and you can add a check for the python version in a future PR if you want to.